libert_util/tests/ert_util_struct_vector.c

```diff
@@ -77,9 +88,56 @@ void test_append_iget() {
 
 
 
+int cmp( const void * _d1, const void * _d2) {
+  const struct test_struct * d1 = (const struct test_struct *) _d1;
+  const struct test_struct * d2 = (const struct test_struct *) _d2;
+
+  if (d1->x > d2->x)
+    return 1;
+
+  if (d1->x < d2->x)
+    return -1;
+
+  return 0;
+}
+
+
+int rcmp( const void * _d1, const void * _d2) {
+  return cmp(_d2 , _d1);
+}
+
+
+
+void test_sort() {
+  struct test_struct d;
+  struct_vector_type * struct_vector = struct_vector_alloc( sizeof d );
+  for (int i = 0; i < 10; i++) {
+    struct test_struct d = {.x = 9 - i,
+                            .y = 9 - i,
+                            .z = 9 - i };
+    struct_vector_append( struct_vector , &d );
+  }
+  struct_vector_sort( struct_vector , cmp );
+  for (int i = 0; i < 10; i++) {
+    struct test_struct d;
+    struct_vector_iget( struct_vector , i  ,&d );
+    test_assert_int_equal( d.x , i );
+    test_assert_int_equal( d.z , i );
```

Comment:
I would just check that they were sorted:
```c
for (int i = 0; i < 9; ++i) {
  struct test_struct left;
  struct test_struct right;
  struct_vector_iget(struct_vector, i, &left );
  struct_vector_iget(struct_vector, i+1, &right );
  test_assert_true(cmp(left, right) >= 0);
}
```

Because that's what we should test: sorted in the eyes of `cmp`.